obs-backscrub.cpp

```diff
@@ -120,7 +121,7 @@ static obs_properties_t *obs_backscrub_get_properties(void *state) {
 }
 static void obs_backscrub_update(void *state, obs_data_t *settings) {
     obs_backscrub_filter_t *filter = (obs_backscrub_filter_t *)state;
-    const char *model = _obs_backscrub_get_model(settings);
+    char *model = _obs_backscrub_get_model(settings);
     obs_printf(filter, "update: model: %s=>%s", filter->modelname, model);
     // here we change any filter settings (eg: model used, feathering edges, bilateral smoothing)
     if (strcmp(model, filter->modelname)) {
```

Comment:
Note that here the `filter->modelname` is being compared to the output of `get_model` - ie it isn't put through `get_path` first. This is actually a bug because `get_path` doesn't change absolute paths, but it does change relative paths. `filter->modelname` always contains the absolute path, because it is always set to the output of `get_path`. So if the setting string is a relative path, this `strcmp` will never match, even if the user didn't change it! Therefore merging the two functions into one actually stealth-fixes a harmless bug.